### PR TITLE
Update debugger documentation

### PR DESCRIPTION
**Pull request by dazuma**
I reviewed and updated the Debugger README, rearranging a few things to provide a better "quickstart" experience, updating to match some recent changes such as renaming the `keyfile` configuration to `credentials`, and adding notes about side effect detection and how to disable it (see https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1875).

### CODE REVIEW

File: google-cloud-debugger/README.md [hunk 1]
```diff
@@ -1,51 +1,41 @@
...
-
-## Enabling the Debugger agent
-
-The Stackdriver Debugger library provides a Debugger agent that helps instrument
-breakpoints in your running applications. The library also comes with a Railtie
-and a Rack Middleware to help control the Debugger agent in popular Rack based
-frameworks, such as Ruby on Rails and Sinatra.
-
-### With Ruby on Rails
+#### Using Ruby on Rails
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Using with Ruby on Rails

File: google-cloud-debugger/README.md [hunk 2]
```diff
@@ -58,7 +48,7 @@ require "google/cloud/debugger/rails"
 If you're using the `stackdriver` gem, it automatically loads the Railtie into
 your application when it starts.
 
-### With other Rack-based frameworks
+#### Using other Rack-based frameworks
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Using with other Rack-based frameworks

File: google-cloud-debugger/README.md [hunk 3]
```diff
@@ -68,38 +58,42 @@ require "google/cloud/debugger"
 use Google::Cloud::Debugger::Middleware
 ```
 
-### Without Rack-based framework
+#### Without a Rack-based framework
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Using without a Rack-based framework

File: google-cloud-debugger/README.md [hunk 4]
```diff
@@ -1,51 +1,41 @@
...
-```sh
-$ bundle install
-```
+Alternatively, consider installing the [`stackdriver`](../stackdriver) gem. It
+includes the `google-cloud-debugger` gem as a dependency, and automatically
+initializes it for some application frameworks.
 
-Alternatively, check out the [`stackdriver`](../stackdriver) gem that includes
-the `google-cloud-debugger` gem.
+### Initializing the debugger
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Inconsistent capitalization. "the Debugger" is capitalized elsewhere.

File: google-cloud-debugger/README.md [hunk 5]
```diff
@@ -68,38 +58,42 @@ require "google/cloud/debugger"
...
 
-The Stackdriver Debugger agent should work without you manually providing
-authentication credentials for instances running on Google Cloud Platform, as
-long as the Stackdriver Debugger API access scope is enabled on that instance.
+To connect to the Stackdriver Debugger service, the agent needs to be
+authenticated. If your application is hosted on Google Cloud Platform, much of
+this is handled for you automatically.
 
-### App Engine
+#### From Google App Engine
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Connecting from Google App Engine (GAE)

File: google-cloud-debugger/README.md [hunk 6]
```diff
@@ -68,38 +58,42 @@ require "google/cloud/debugger"
...
 
-On Google App Engine, the Stackdriver Debugger API access scope is enabled by
-default, and the Stackdriver Debugger agent can be used without providing
-credentials or a project ID.
+If your app is running on Google App Engine, the Stackdriver Debugger agent
+authenticates automatically by default, and no additional configuration is
+required.
 
-### Container Engine
+#### From Google Kubernetes Engine
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Connecting from Google Kubernetes Engine (GKE)

File: google-cloud-debugger/README.md [hunk 7]
```diff
@@ -108,11 +102,14 @@ $ gcloud container clusters create example-cluster-name --scopes https://www.goo
...
 **Enabled** in the Cloud Platform section of **Create a container cluster**.
 
-### Compute Engine
+After the OAuth scope is enabled, the Stackdriver Debugger agent authenticates
+automatically by default, and no additional configuration is required.
 
-To use Stackdriver Debugger, Compute Engine VM instances should have one of the
-following access scopes. These are only relevant when you use Compute Engine's
-default service account:
+#### From Google Compute Engine
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Connecting from Google Compute Engine (GCE)

File: google-cloud-debugger/README.md [hunk 8]
```diff
@@ -184,15 +177,58 @@ This library also supports the other authentication methods provided by the
...
+By default, the snapshot includes the local variables from the current and four
+most recent stack frames. You may include additional data in the snapshot by
+providing a list of _expressions_ when you create the breakpoint. Expressions
+may be instance variables, global variables, or the result of calling Ruby
+methods, or indeed, any Ruby expression.
+
+For more information on using the debugger features, see the
+[Stackdriver Debugger Documentation](https://cloud.google.com/debugger/docs/).
+
+#### Mutation protection
```
Comment (quartzmo) [inline google-cloud-debugger/README.md]: Disabling mutation protection

**Comment by blowmage** [comment]
Great improvement!

**Pull request merged**